lib/Service/CollectiveShareService.php

```diff
@@ -156,32 +197,46 @@ private function isShareEditable(IShare $folderShare): bool {
 	/**
 	 * @param string         $userId
 	 * @param CollectiveInfo $collective
+	 * @param PageInfo|null  $pageInfo
 	 *
 	 * @return CollectiveShareInfo
 	 * @throws NotFoundException
 	 * @throws NotPermittedException
 	 */
-	public function createShare(string $userId, CollectiveInfo $collective): CollectiveShareInfo {
+	public function createShare(string $userId, CollectiveInfo $collective, ?PageInfo $pageInfo): CollectiveShareInfo {
 		if (!$collective->canShare()) {
 			throw new NotPermittedException($this->l10n->t('You are not allowed to share %s', $collective->getName()));
 		}
 
-		if (null !== $this->findShare($userId, $collective->getId())) {
+		$pageId = 0;
+		$nodeId = 0;
+		if ($pageInfo) {
+			$pageId = $pageInfo->getId();
+			$file = $this->pageService->getPageFile($collective->getId(), $pageId, $userId);
+			if (NodeHelper::isIndexPage($file)) {
+				$nodeId = $file->getParent()->getId();
+			} else {
+				$nodeId = $pageInfo->getId();
+			}
```

Comment:
I might be confusing things here. I thought you intended to turn normal `page.md` pages into a `Readme.md` within their own folder so you can share the folder and thus all future subpages.

This code looks more like you're handling the pages without pages just like any other file.

Reply:
Oh, good catch. This if condition can go. We already make sure that the `pageInfo` passed to `createShare` isn't a single page file anymore [here](https://github.com/nextcloud/collectives/blob/388224b3e9815653747995a226313dea677b7c8f/lib/Controller/ShareController.php#L126).